### PR TITLE
Fix and rewrite audio and subtitle track selection when transcoding

### PR DESCRIPTION
**Pull request by Maxr1998**

### CODE REVIEW

File: app/src/main/java/org/jellyfin/mobile/player/ui/PlayerFragment.kt [hunk 1]
```diff
@@ -262,30 +264,34 @@ class PlayerFragment : Fragment() {
...
 
     /**
-     * @return true if the audio track was changed
+     * @param callback called if track selection was successful and UI needs to be updated
      */
-    fun onAudioTrackSelected(index: Int): Boolean {
-        return viewModel.selectAudioTrack(index)
+    fun onAudioTrackSelected(index: Int, callback: TrackSelectionCallback): Job = lifecycleScope.launch {
+        if (viewModel.selectAudioTrack(index)) {
+            callback.onTrackSelected(true)
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/mobile/player/ui/PlayerFragment.kt:271]: Why not make this suspended instead of using a callback?

Comment (nielsvanvelzen) [reply app/src/main/java/org/jellyfin/mobile/player/ui/PlayerFragment.kt:271]: Same question for the other methods

File: app/src/main/java/org/jellyfin/mobile/player/source/MediaSourceResolver.kt [hunk 2]
```diff
@@ -33,6 +33,10 @@ class MediaSourceResolver(private val apiClient: ApiClient) {
                 itemId = itemId,
                 data = PlaybackInfoDto(
                     userId = apiClient.userId,
+                    // We need to remove the dashes so that the server can find the correct media source.
+                    // And if we didn't pass the mediaSourceId, our stream indices would silently get ignored.
+                    // https://github.com/jellyfin/jellyfin/blob/9a35fd673203cfaf0098138b2768750f4818b3ab/Jellyfin.Api/Helpers/MediaInfoHelper.cs#L196-L201
+                    mediaSourceId = itemId.toString().replace("-", ""),
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/mobile/player/source/MediaSourceResolver.kt:39]: This conversion feels hacky, do we not have a reference to the mediasource so we can use the id from that?

**Review by nielsvanvelzen — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/jellyfin/mobile/player/source/MediaSourceResolver.kt (hunk 2 above)
Comment (Maxr1998) [reply app/src/main/java/org/jellyfin/mobile/player/source/MediaSourceResolver.kt:39]: It is hacky but sadly not. The id shouldn't be a string anyways but the api model would require changes for this.

File: app/src/main/java/org/jellyfin/mobile/player/ui/PlayerFragment.kt (hunk 1 above)
Comment (Maxr1998) [reply app/src/main/java/org/jellyfin/mobile/player/ui/PlayerFragment.kt:271]: I didn't want to add coroutines to the UI part

File: app/src/main/java/org/jellyfin/mobile/player/source/MediaSourceResolver.kt (hunk 2 above)
Comment (nielsvanvelzen) [reply app/src/main/java/org/jellyfin/mobile/player/source/MediaSourceResolver.kt:39]: @crobibero can you help us out here, would it be possible to 1) change the parameter type to a UUID, or 2) allow a UUID with dashes to be auto-converted serverside?

File: app/src/main/java/org/jellyfin/mobile/player/ui/PlayerFragment.kt (hunk 1 above)
Comment (nielsvanvelzen) [reply app/src/main/java/org/jellyfin/mobile/player/ui/PlayerFragment.kt:271]: We can use the viewmodel lifecycle so I don't really see the issue.

Comment (Maxr1998) [reply app/src/main/java/org/jellyfin/mobile/player/ui/PlayerFragment.kt:271]: Need to check if PlayerMenus has that. But it's an option, true.

File: app/src/main/java/org/jellyfin/mobile/player/source/MediaSourceResolver.kt (hunk 2 above)
Comment (crobibero) [reply app/src/main/java/org/jellyfin/mobile/player/source/MediaSourceResolver.kt]: Which parameter do you want changed? `MediaSourceInfo`?

Comment (Maxr1998) [reply app/src/main/java/org/jellyfin/mobile/player/source/MediaSourceResolver.kt]: The `mediaSourceId`, but I don't know whether it supports more than UUIDs.

**Review by nielsvanvelzen — APPROVED** [review]

**Pull request merged**